### PR TITLE
Labels to be added to JIRA tickets parametrized

### PR DESCRIPTION
**Pull request by trepel**
*Motivation*

The 'test-case' label were hard-coded. It is much better to parametrize this since there is no reason for thinking each JIRA ticket always needs 'test-case'  label on it.

*How to test*
`trira sync 'Test Plan: Self Managed' 'TEST-34' --list-regexp='MUST PER RELEASE' '--card-regexp=RHMAP proxy' --debug --dry-run --labels=test-case,rhmap-signoff

You can experiment with value for `--labels` and remove `--dry-run`. Then see https://issues.jboss.org/browse/TEST-34 epic

### CODE REVIEW

**Comment by trepel** [comment]
@jhellar, @matskiv I cannot ask you to review this. And I cannot merge this either. Do you have rights to LGTM (and merge) this ?

File: cmd/sync.js
Comment (kpiwko) [inline cmd/sync.js]: @trepel this was problem with previous implementation as well - but I can see that while labels are unique, here we are able to more easily define one label twice - I'm not use if JIRA won't choke on that

Comment (trepel) [reply cmd/sync.js]: @kpiwko My understanding of your comment is that the code does not handle the situation when the labels array contain duplicates. Thanks for pointing this out, I haven't thought about it to be honest.

I tried and JIRA handles such situation well - no errors, only one label (regardless the number of duplicates) is added to JIRA ticket. However if using --dry-run, you can see all the duplicates in the output. I will fix it by removing duplicates.

Comment (trepel) [reply cmd/sync.js]: I tried to handle duplicities and also empty strings. Please re-review if you have time. Thanks.

**Comment by trepel** [comment]
@kpiwko Updated, does it look good to you now ?

File: cmd/sync.js
Comment (kpiwko) [inline cmd/sync.js]: This is wrong - what if I don't want to push any extra labels @trepel? And it also duplicates default label at 2 places.

File: cmd/sync.js
Comment (kpiwko) [inline cmd/sync.js:153]: This might be reduced to a single line a guess but in sake of legibility +1

File: cmd/sync.js
Comment (trepel) [reply cmd/sync.js]: I left default value because there were default value in previous implementation (I just changed it from 'test-case' to 'trira'. You are right that there is no way to push no label. I will remove the default value altogether.

File: cmd/sync.js
Comment (trepel) [reply cmd/sync.js:153]: Yes, this is more philosophical question to me. Successful code has very long life cycle where initial development is actually very short part. The maintenance part takes much longer.  Having a lot of stuff done on one line using just few characters is great for developer (nicer, less typing) but not so convenient for maintainer (more difficult to understand and debug). On the other hand some say that less characters means less opportunities for bugs. Not sure where the balance is.

In this particular case it is not super-complex thus it could have been reduced, but since I was not familiar with the '...' operator I did split it and added the comment. Since you +1'ed it, I leave it as it is.

**Comment by trepel** [comment]
@kpiwko Thanks for review. You can re-review again now.

**Comment by kpiwko** [comment]
I know I'm very annoying, sorry @trepel  :-)

What would you expect this command to do?
```
trira sync 'Test Plan: Deploy' RHMAP-123456 --labels 'foo,bar'
```
I would say all cards with given labels

Maybe it should be called `--add-labels` instead ?

**Comment by trepel** [comment]
no worries @kpiwko , I got used to it over the years :-)

I was aware of the fact that you would prefer '--add-labels' because you mentioned something similar in https://github.com/aerogear/trira/issues/5. I didn't name it that way because other arguments (--list-regexp, --card-regexp, --dry-run) were nouns so I thought that using just '--labels' might be better since it is noun too.

Happy to change it to '--add-labels' though

**Comment by trepel** [comment]
Hi @kpiwko ,
it is more or less ready for next review. Thanks for your time and effort.

Known issues: it is not possible to add `false` string as label. It is not possible to add string consisting just of one or more `0` (zero) characters as label.

File: cmd/sync.js
Comment (kpiwko) [inline cmd/sync.js]: @trepel known issues is because of here should be a smarter check - I would say default value for addLabes undefined and then === check for undefined and false and 0 would be possible? ;-)

Comment (kpiwko) [reply cmd/sync.js]: And now I'm really enjoying being annoying 🤣

**Comment by trepel** [comment]
Ha, that (undefined as default value) was pretty good idea @kpiwko. Done. There are still some caveats though. Two or more zero will be interpreted as just one zero. In general,  leading zeros will be ignored. Furthermore, `0x<somenumber>` will be interpreted as hexadecimal number before converting to string. Thus `--add-labels=0xA` will cause that label `10` is added.

And if you use just `--add-labels` (without `=` and any value) then 'true' label will be added.

**Comment by trepel** [comment]
@kpiwko  is the way it is now acceptable, wdyt ?

**Review by kpiwko — APPROVED** [review]
LGTM.

**Pull request merged**
